packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -120,6 +120,63 @@ module.exports = {
     }
   },
 
+  async changePassword(ctx) {
+    const params = _.assign({}, ctx.request.body, ctx.params);
+
+    if (!_.get(ctx, 'state.user.id')) {
+      throw new ApplicationError('You must be authenticated to reset your password');
+    }
+
+    if (
+      params.currentPassword &&
+      params.password &&
+      params.passwordConfirmation &&
+      params.passwordConfirmation === params.password
+    ) {
+      const user = await strapi.entityService.findOne(
+        'plugin::users-permissions.user',
+        ctx.state.user.id
+      );
+
+      if (!user) {
+        throw new ValidationError('User not found');
+      }
+
+      const validPassword = await getService('user').validatePassword(
+        params.currentPassword,
+        user.password
+      );
+
+      if (!validPassword) {
+        throw new ValidationError('Your current password is invalid');
+      }
+
+      if (params.currentPassword === params.password) {
+        throw new ValidationError('Your new password must be different than your current password');
+      }
+
+      await getService('user').edit(user.id, {
+        resetPasswordToken: null,
+        password: params.password,
+      });
+
+      ctx.send({
+        jwt: getService('jwt').issue({ id: user.id }),
```

Comment:
I'm wondering if it makes sense to refresh your jwt token when changing password 🤔 do you see some value into it ?

Reply:
Would we have any way to invalidate JWTs obtained before the password change?
If so, providing a new JWT would make sense so that the user doesn't have to log in again, but if we don't, then I guess it doesn't really make sense.

Reply:
yes we should invalidate the JWT, but you are correct @WalkingPizza we have no way to invalidate them. :sob: 

It means we would need to encrypt an additional option into the JWT check that but we can't encrypt the password into the JWT itself. I think it falls out of scope in the current U&P plugin @alexandrebodin and likely just needs to be something we consider in it's replacement we have been talking about.

cc @marcoautiero